### PR TITLE
Fix personalization modal properties locale reactivity

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | If we open a shoe with locale set to `en_us` then change that locale to `ja_jp` in white technical settings, the personlization modal will not update it's property options. See GIF below |
| Decisions | `locale_map_changed` needs to be triggered every time `localeMap` suffers mutations. Add triggers after calling `loadModelLocales` and  `loadRipeSdkLocales`.  |
| Animated GIF | Below |

### Before
Peek 2020-11-19 18-13

### After
Peek 2020-11-19 18-14

### CODE REVIEW

**Review by gcandal — APPROVED** [review]

File: js/locale/locale.js [hunk 1]
```diff
@@ -53,14 +54,15 @@ export class LocalePlugin extends RipeCommonsPlugin {
...
-        return this.getLocaleValue(key, defaultValue, locale, fallback);
+    toLocale(key, defaultValue, locale, fallback = true, localeMap) {
+        return this.getLocaleValue(key, defaultValue, locale, fallback, localeMap);
     }
 
     async setLocale(locale, coerce = true) {
         if (coerce) locale = this._coerceLocale(locale);
         await this.owner.trigger("pre_set_locale", locale);
         this.locale = locale;
+        this.localeMap = await this._loadLocales();
```
Comment (gcandal) [inline js/locale/locale.js:65]: Why is this needed?

Comment (3rdvision) [reply js/locale/locale.js:65]: This is needed to assign the new `localeMap` when a different locale is set.
With this, `localeMap` change will trigger the computed `locale()` which fixes the bug

Comment (gcandal) [reply js/locale/locale.js:65]: I still don't get why, because this function changes `this.locale = locale` and we're already taking care of that in:

```
localePlugin.owner.bind("post_set_locale", locale => (this.currentLocale = locale));
```

Let's go over this on Discord. Please ping me

Comment (3rdvision) [reply js/locale/locale.js:65]: I still think we need this line after your suggestion https://github.com/ripe-tech/ripe-commons-pluginus/blob/master/vue/plugins/locale.js#L37
post_set_locale is only setting the currentLocale with old `localeMap`.
Only after `await _loadLocales` `localeMap` changes and is able to trigger it.

Closing and opening modal works without this line but it also works in master.
I believe this fix could potentially solve the need to `forceUpdate` in reference and pickers

Comment (gcandal) [reply js/locale/locale.js:65]: Before this line, when did the `localeMap` get updated?

Comment (3rdvision) [reply js/locale/locale.js:65]: in `loadModelLocales` https://github.com/ripe-tech/ripe-commons-pluginus/blob/master/js/locale/model-locale-loader.js#L24

I am going to refactor this solution as some things have changed ever since this was proposed and it's not the best solution anymore as part of the fix was done outside of this PR but not all of it.

Comment (gcandal) [reply js/locale/locale.js]: Much better now!

**Review by gcandal — APPROVED** [review]
